rfcs/text/0045-additional-vulnerability-fields.md

```diff
@@ -0,0 +1,138 @@
+| `vulnerability.mitigation` | text | Explains user how to fix or mitigate the problem, could be usefd to store resolution from the scanner vendor or document mitigation in place |
+| `vulnerability.published`	| date | The “published” field indicates the date when information about a specific vulnerability was publicly disclosed or made available.It represents the moment when details about the vulnerability were shared with the security community, vendors, and the public.This field helps security professionals track the timeline of vulnerability awareness.Example value: If a critical vulnerability (e.g., CVE-2024-1234) affecting a widely used software is disclosed on February 1, 2024, the published date for that CVE would be February 1, 2024. |
+| `vulnerability.patch.*` | object | - |
+| `vulnerability.patch.exists` | boolean | The “patch” field refers to whether a security fix or update (commonly known as a patch) is available to address the identified vulnerability. It indicates whether the software vendor or developer has released a solution to mitigate the vulnerability. |
+| `vulnerability.patch.name` | text | Name of the patch |
+| `vulnerability.patch.code` | keyword | Associated patch code for example ESA-2020-13 |
+| `vulnerability.evidence` | text | A demonstration of the validity of a vulnerability claim, e.g. app.any.run replaying the exploitation of the vulnerability. |
+| `vulnerability.status` | keyword | The status field helps security teams track vulnerabilities, prioritize actions, and communicate their progress effectively |
+| `vulnerability.tags` | keyword | This is different from cloud provider assigned resource tags, this is specifically for vulnerability. Vulnerability tags serve as a way to add custom metadata to vulnerabilities, enhancing their context and aiding in search and automation. |
+| `vulnerability.first_found` | date | First time a vulnerability was found on the asset. |
+| `vulnerability.last_found` | date | Last time a vulnerability was found on the asset.
+| `vulnerability.last_scanned` | date | Last time a scan was performed on the asset. It's important as some companies are scanning on a quarterly basis. If lastfound and lastscanned are close, it means it's still an active vulnerability. |
+| `vulnerability.age` | date | Numbers of days since the vulnerability is active. It should be dynamically calculated (runtime fields, ingest, ...). It could either be then difference between the lastfound date and the published date (preferred). It could also be the difference between the firstfound and lastfound dates. |
+| `vulnerability.uid` | keyword | It's extremely import to be able to deduplicate different scans. It's often that we have different scanners showing the same vulnerability on the same asset. |
+| `vulnerability.type` | keyword | To conclude it the vulnerability is confirmed or potential. |
```

Comment:
```suggestion
| `vulnerability.type` | keyword | To conclude if the vulnerability is confirmed or potential. |
```

are `confirmed` and `potential` the examples here? Do we have a predefined set of types?

Reply:
Confirmed and Potential are predefined set types.